### PR TITLE
#78 - Customize mail - Backend

### LINKED ISSUE #78: Customize mail -  Backend

### PR DESCRIPTION
**Pull request by PrabuckiDominik**
This PR should close #78 .

### CODE REVIEW

File: app/Providers/AppServiceProvider.php [hunk 1]
```diff
@@ -16,5 +18,13 @@ public function register(): void
     public function boot(): void
     {
         JsonResource::withoutWrapping();
+        VerifyEmail::toMailUsing(function (object $notifiable, string $url) {
+            return (new MailMessage())
+                ->subject("Weryfikacja Adresu E-mail")
```
Comment (EwelinaSkrzypacz) [inline app/Providers/AppServiceProvider.php:23]:
```suggestion
                ->subject("Weryfikacja adresu e-mail")
```

File: resources/views/emails/auth/reset-password.blade.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    <title>Zresetuj hasło</title>
+</head>
+
+<header>
+    <img src="favicon.png" alt="{{ config('app.name') }} Logo" width="150" height="50">
+    <h1>{{ config('app.name') }}</h1>
+</header>
+
+<body>
+<h1>Witaj {{ $user->name }},</h1>
```
Comment (EwelinaSkrzypacz) [inline resources/views/emails/auth/reset-password.blade.php:16]:
```suggestion
<h1>Cześć {{ $user->name }},</h1>
```

File: resources/views/emails/auth/reset-password.blade.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+<p>Otrzymaliśmy prośbę o zresetowanie hasła dla Twojego konta.</p>
+
+<p>Aby zresetować swoje hasło, kliknij poniższy link:</p>
+
+<a href="{{ $url }}">Zresetuj hasło</a>
+
+<p>Jeśli to nie Ty wysłałeś/aś tę prośbę, po prostu zignoruj tę wiadomość.</p>
```
Comment (EwelinaSkrzypacz) [inline resources/views/emails/auth/reset-password.blade.php:24]:
```suggestion

```

File: resources/views/emails/auth/verify.blade.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    <title>Witaj w {{ config('app.name') }}</title>
+</head>
+
+<header>
+    <img src="" alt="{{ config('app.name') }} Logo" width="150" height="50">
+    <h1>{{ config('app.name') }}</h1>
+</header>
+
+<body>
+<p>Witaj {{ $user->name }},</p>
```
Comment (EwelinaSkrzypacz) [inline resources/views/emails/auth/verify.blade.php:16]:
```suggestion
<p>Cześć {{ $user->name }},</p>
```

File: resources/views/emails/auth/verify.blade.php [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+<p>Aby zweryfikować swoje konto, kliknij poniższy link:</p>
+
+<a href="{{ $url }}">Zweryfikuj E-mail </a>
```
Comment (EwelinaSkrzypacz) [inline resources/views/emails/auth/verify.blade.php:20]:
```suggestion
<a href="{{ $url }}">Zweryfikuj e-mail </a>
```

File: resources/views/emails/auth/verify.blade.php [hunk 6]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Witaj w {{ config('app.name') }}</title>
```
Comment (EwelinaSkrzypacz) [inline resources/views/emails/auth/verify.blade.php:7]:
```suggestion
    <title> {{ config('app.name') }}</title>
```

File: app/Notifications/ResetPasswordNotification.php [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+{
+    use Queueable;
+
+    public function __construct(
+        protected string $token,
+    ) {}
+
+    /**
+     * @return array<int, string>
+     */
```
Comment (EwelinaSkrzypacz) [inline app/Notifications/ResetPasswordNotification.php:21]:
```suggestion
```

**Review by EwelinaSkrzypacz — APPROVED** [review]

File: app/Providers/AppServiceProvider.php [hunk 8]
```diff
@@ -16,5 +18,13 @@ public function register(): void
     public function boot(): void
     {
         JsonResource::withoutWrapping();
+        VerifyEmail::toMailUsing(function (object $notifiable, string $url) {
```
Comment (AleksandraKozubal) [inline app/Providers/AppServiceProvider.php:21]: Refactor the callback to use a shorter arrow function (fn) with a return type hint. This makes the code more concise and ensures type safety by explicitly returning a MailMessage object

File: resources/views/emails/auth/verify.blade.php [hunk 9]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
+<html lang="en">
```
Comment (AleksandraKozubal) [inline resources/views/emails/auth/verify.blade.php:2]:
```suggestion
<html lang="pl">
```

**Review by AleksandraKozubal — APPROVED** [review]

**Review by EwelinaSkrzypacz — APPROVED** [review]

**Pull request merged**